server/app/filters/ApiKeyUsageFilter.java

```diff
@@ -0,0 +1,74 @@
+package filters;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import auth.ProfileUtils;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import play.api.libs.concurrent.AkkaSchedulerProvider;
+import play.mvc.EssentialAction;
+import play.mvc.EssentialFilter;
+import play.mvc.Http;
+import scala.concurrent.ExecutionContext;
+import services.apikey.ApiKeyService;
+
+/**
+ * This filter looks for requests with paths that begin with /api and record the usage data for the
+ * relevant API key if it has one. The API key update is scheduled to run in a background through to
+ * reduce latency and to ensure issues with recording the usage do not cause API requests to fail.
```

Comment:
Not blocking: I'd expand the comment either here or where we make the background call noting that both eventual consistency as well as potentially failed increments are both acceptable (the latter in the case of many concurrent API calls occurring in short succession).

Also, to confirm, lots of API key calls updating the database could lead to DB contention. Presumably, that's acceptable and not something we're worrying about for V1?

Reply:
> Also, to confirm, lots of API key calls updating the database could lead to DB contention. Presumably, that's acceptable and not something we're worrying about for V1?

yep, not a concern for the foreseeable future